src/components/nav/desktop/NavAudio.tsx

```diff
@@ -44,45 +43,24 @@ const NavAudio = () => {
   const { tier } = useSelectTierInfo(account?.user_id ?? 0)
   const audioBadge = audioTierMapPng[tier]
 
-  // Pending claim logic
-  const pendingAutoClaims = useSelector(getPendingAutoClaims)
-  const [hasExpiredClaim, setHasExpiredClaim] = useState(false)
-  const [bubbleType, setBubbleType] = useState<BubbleType>('empty')
-  const claimPromptDelayMs = useRemoteVar(IntKeys.MANUAL_CLAIM_PROMPT_DELAY_MS)
+  const userChallenges = useOptimisticUserChallenges()
+  const hasExpiredClaim = Object.values(userChallenges).some(
+    challenge => challenge?.state === 'completed'
+  )
```

Comment:
Feels like this guy wants to be it's own selector perhaps

Reply:
Given that it's using the return value of a hook, maybe not?

Reply:
fair point, just felt odd to fetch all the challenges and then just use them to compute this extra state